src/corppa/poetry_detection/core.py

```diff
@@ -79,6 +81,49 @@ def overlap_factor(self, other: "Span", ignore_label: bool = False) -> float:
         return overlap / max(len(self), len(other))
 
 
+def field_real_type(field_type) -> type:
+    """Return the base type for a dataclass field type annotation.
+    For unions or optional values, returns the first non-None type; for type
+    aliases, returns the original type.
+    """
+    # if it's a regular type, return unchanged
+    if isinstance(field_type, type):
+        return field_type
+    # for a type alias, return the original type
+    # e.g. for annotation of set[str] return the set type
+    origin = get_origin(field_type)
+    if isinstance(origin, type):
+        return origin
+    # if Optional or Union, return the first non-none type
+    ftypes = get_args(field_type)
+    if ftypes:
+        return [arg for arg in ftypes if arg != types.NoneType][0]
+
+    # if we get here, this is an input we can't handle
+    raise TypeError(f"Cannot determine real type for '{field_type}'")
+
+
+#: character to use when converting sets to and from delimited string
+MULTIVAL_DELIMITER = ";"
```

Comment:
Since the delimiter is presumably being used in fields that might be read (i.e., CSVs); can this be a two-character delimiter?
```suggestion
MULTIVAL_DELIMITER = "; "
```